test/Microsoft.VisualStudio.Threading.Tests/JoinableTaskTests.cs

```diff
@@ -3835,6 +3835,244 @@ public void JoinableTaskOfT_TaskPropertyBeforeReturning()
         });
     }
 
+    [Fact]
+    public void IsCompletedTrueDoesNotLock()
+    {
+        using var context = new JoinableTaskContext();
+        var syncContextLock = typeof(JoinableTaskContext).GetProperty("SyncContextLock", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(context);
```

Comment:
Using reflection to access private fields violates the rule that leads to our no-IVT policy for tests. It makes for a less compelling test when you can't assert that it's a publicly observable functional difference. Can we instead find a deadlock risk to exploit, or whatever else motivated you to make this change in the first place, and code that up using only public API?

Reply:
There is no deadlock risk. This is a performance test which covers an implementation detail not exposed through the API.